src/RFC-1102_TappletsRegistry.md

```diff
@@ -0,0 +1,263 @@
+Tapplets themselfs are packages kept in any registry, like npm.
+
+## npm registry
+
+Let [MetaMask Snaps](https://docs.metamask.io/snaps/learn/about-snaps/) serve as an example.
+Github repository is the place where docs, examples, manifest files and so on are stored. Let's assume that every tapplet is an npm package published for specified workspace as e.g. “tari-universe”.
+
+During the discussion about this RFC, it was correctly pointed out that one possible drawback of npm registry is the risk that dependencies will introduce vulnerabilities, which is known as *supply chain attack*. However, by design, tapplets must not have any dependencies required for installation (which should be checked in the add/update process) and must be prepared as production-ready bundles.
+
+To sum up:
+
+- npm stores tapplets
+- GitHub repository is the Tapplet Registry and stores tapplets list
+
+## Scenarios
+
+### Adding a new tapplet
+
+Creating PR to the Github Repo by the tapplet author.
+Approval should be done only if the tapplet is checked and verified
+
+- Github flow
+
+  1. Fork Tapplets Registry repo and create new branch with a new tapplet files. Add required `tapplet.manifest.json` file and readme.
+  2. Sign with GPG key
+  3. Register the tapplet:
+
+  - create pull request
+  - CLI provides the tapplet data (taken from the package)
+  - Github Actions verify - redo previous step to verify if provided data is valid
+  - Add GitHub CODEOWNERS:
+    - Allows the Tapplet’s contributor for future updates
+    - Restrict Tapplet Registry repo from being modified by tapplets contributors
+
+- Tari Blockchain flow
+
+  1. Compress a tapplet project to zip file.
+  2. Generate checksum and and sign the blockchain transaction.
+  3. Call contract's function, like registerTapplet() which verifies signature and add the checksum to the verified tapplets mapping.
+
+- npm package registry flow
+  1. Create the tapplet package with required files
+  2. Publish to npm
+  3. Register the tapplet
+  - Github PR (GPG key used mandatory)
+  - CLI provides the tapplet data (taken from the package)
+  - Github Actions verify - redo previous step to verify if provided data is valid
+  - Add GitHub CODEOWNERS:
+    - Allows the Tapplet’s contributor for future updates
+    - Restrict Tapplet Registry repo from being modified by tapplets contributors
+
+### Updating a tapplet version
+
+The upgrade process is almost the same as creating a new one, but with PR only changes need to be checked.
+
+### Deprecating/removing a tapplet
+
+It may (and pretty sure will) happen that a specific version is not recommended for use because of a bug or a business decision. In that case following options are considered:
+
+1. Mark the version as “deprecated” - special optional tag like “status” can be used
+2. Update the Tapplet Registry and remove the tapplet from the list available ones
```

Comment:
Ok, I asked about this in RFC-1101. 
So, there's a high degree of trust being placed in the maintainers of the registry repo. 

It would be good to make this trust model explicit somewhere.